src/charm.py

```diff
@@ -905,11 +906,11 @@ def update_config(self) -> None:
 
         # Update and reload configuration based on TLS files availability.
         self._patroni.render_patroni_yml_file(
-            archive_mode=self.app_peer_data.get("archive-mode", "on"),
             connectivity=self.unit_peer_data.get("connectivity", "on") == "on",
             enable_tls=enable_tls,
             backup_id=self.app_peer_data.get("restoring-backup"),
             stanza=self.app_peer_data.get("stanza"),
```

Comment:
on restore we put an empty string `""` into relation, what value do we get here? empty string or None?
the problem here, is that in `render_patroni_yml_file`, depending on the value we enable pgbackrest as an archive command if `enable_pgbackrest=stanza is not None`.
but in python empty string is not None